tensorflow/compiler/xla/service/platform_util.h

```diff
@@ -64,7 +64,8 @@ class PlatformUtil {
   //
   // If the platform has no visible devices, a not-found error is returned.
   static StatusOr<std::vector<se::StreamExecutor*>> GetStreamExecutors(
-      se::Platform* platform, std::set<int> allowed_devices = {-1});
+      se::Platform* platform,
+      const absl::optional<std::set<int>>& allowed_devices = {});
```

Comment:
Please comment what the `allowed_devices` argument does and why it's there.  (That is, why can't we just take the resulting vector and *then* filter it?  That's the whole point of this patch, actually; it's quite subtle, and right now it's not explained anywhere in the code.)